### ISSUE TITLE
Self host

### ISSUE THREAD

**Issue by AykhanUV** [himanshu8443/8-Stream#7]
Hey can i self host this frontend? Also can you please teach me how can i change the streaming provider api's

**Comment by himanshu8443** [comment]
Yes, you can self-host as it's using Next.js 14. However, it wasn't designed with the intention of utilizing additional APIs, so you would need to refactor some components in src/components/watch.

**Comment by AykhanUV** [comment]
There are 3 files in src/components/watch but i did not find api in there can u please explain me? I want to change api to vidsrc.pro one

**Comment by himanshu8443** [comment]
I didn't mean that you only need to change those three files. Those files also import functions from other files and also check in components/Player/Stream.tsx.
"Furthermore, this project is currently not being actively developed."

**Comment by AykhanUV** [comment]
i just wanna change the provider, can u just tell me which file api is in

**Comment by himanshu8443** [comment]
src/lib folder contains all API call functions
